docs/howto/source.rst

```diff
@@ -0,0 +1,144 @@
+.. _metricq-source-ipmi: https://github.com/metricq/metricq-source-ipmi
+.. _metricq-source-bacnet: https://github.com/metricq/metricq-source-bacnet
+
+Here, we will build a simple Source that sends a random value between 0 and 1 at a configurable interval.
+
+Defining a new Source: using :class:`metricq.Source`
+----------------------------------------------------
+
+Similarly to Sinks, we create a new Source by subclassing :class:`metricq.Source`:
+
+.. code-block:: python
+
+    import metricq
+
+    class DummySource(metricq.Source):
+        pass
+
+The Source will receive its configuration dynamically over the network.
+This is different from the Sink we built in :ref:`sink-how-to`;
+we won't pass the rate at which new values are sent on the command line,
+instead we install a callback that is triggered every time a new configuration is received:
+
+.. code-block:: python
+
+    import metricq
+
+    class DummySource(metricq.Source):
+        def __init__(self, *args, **kwargs):
+            # This will be set to the number of values we want to send per second:
+            self._rate = None
+            super().__init__(*args, **kwargs)
+
+        @metricq.rpc_handler("config")
+        async def _on_config(self, **config):
+            print(f"Received new configuration: {config}")
+            self._rate = config["rate"]
+
+Clients on the MetricQ network communicate via an `RPC protocol <https://metricq.github.io/metricq-rpc-docs/>`_.
+The :func:`metricq.rpc_handler` decorator is a way to define a new handler for an RPC;
+here we tell the library to call :code:`_on_config` every time another client sends a :literal:`"config"`-RPC
+which contains our new configuration.
+
+So far out Source would work, but it wouldn't do anything useful at all.
+To change that, we first declare for which metric we want to send values:
+
+.. code-block:: python
+
+    import metricq
+
+    class DummySource(metricq.Source):
+        ... # as above
+
+        @metricq.rpc_handler("config")
+        async def _on_config(self, **config):
+            print(f"Received new configuration: {config}")
+            self._rate = config["rate"]
+
+            metadata = {
+                "rate": rate,
+                "description": "A simple dummy metric providing random values, sent from a python DummySource",
+            }
+
+            await self.declare_metrics({"example.py.dummy": metadata})
+
+If we were in a hurry, we could omit the metadata for our metric,
+see :meth:`Source.declare_metrics` on how to do that.
+
+To finally send some values, we override :meth:`Source.task`.
+This method gets called once our Source is connected and received its initial configuration:
+
+.. code-block:: python
+
+    import metricq
+    import asyncio
+    import random
+
+    class DummySource(metricq.Source):
+        ... # as above
+
+        async def task(self):
+            while True:
+                await self.send(
```

Comment:
Please mention that users need to save the task (running `task()`) from being canceled in case of a reconnect.
See https://github.com/metricq/metricq-python/commit/798282e671b71359b2192b9db7fd125d8183e646

Reply:
Done in f78b51e1e.

Is this what you had in mind?